### PR TITLE
feat: init gio-search-menu component

### PR DESCRIPTION
**Pull request by Okhelifi**
**Description**

add `gio-search-menu` component for the side nav

**Additional context**

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-particles-angular
```
npm install @gravitee/ui-particles-angular@9.2.0-apim-3442-menu-search-ecbb5ef
```
```
yarn add @gravitee/ui-particles-angular@9.2.0-apim-3442-menu-search-ecbb5ef
```

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-policy-studio-angular
```
npm install @gravitee/ui-policy-studio-angular@9.2.0-apim-3442-menu-search-ecbb5ef
```
```
yarn add @gravitee/ui-policy-studio-angular@9.2.0-apim-3442-menu-search-ecbb5ef
```

---

### CODE REVIEW

**Comment by Okhelifi** [comment]
yarn test works locally but keep failing on the CI 🤷🏼 🤔 
Capture d’écran 2024-02-09 à 10 15 43

File: ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      startWith(''),
+      map(value => {
+        const name = typeof value === 'string' ? value : value?.name;
+        return this._filter(name);
+      }),
+    );
+  }
+
+  protected displayFn(item: MenuSearchItem): string {
+    return item && item.name ? item.name : '';
```
Comment (jgiovaresco) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts:56]: This should also work. But I'll let you decide if it worth the change 🙈 

```suggestion
    return item?.name ?? ''
```

File: ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  protected onSelectionChange(event: MatAutocompleteSelectedEvent) {
+    this.control.reset();
+    this.ngOnInit();
```
Comment (jgiovaresco) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts:61]: I don't understand why this call is necessary. The reset should be enough 🤔

Comment (Okhelifi) [reply ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts]: Done. I've updated and tested it on APIM side

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**